Note/VC/ModalViewController.swift

```diff
@@ -53,11 +53,15 @@ class ModalViewController: UIViewController, UITextViewDelegate{
       textView.textColor = UIColor.lightGray
     }
   }
-    
+  
   @IBAction func completionButton(_ sender: Any) {
     let count: Int = ObjectStore.shared.objects.count
     if textView.text != "Нотатки" && (textView.text != nil) != textView.text.isEmpty{
-      ObjectStore.shared.add(note: Note(id: count, name: textView.text, isDone: defaultValue, deadlineDate: dataValue))
+      if statusData.isOn{
```

Comment:
краще зробити так, щоб цей if рахував значення дати, і потім ти один раз звертався до objectStore

Reply:
тут не погоджуюсь, бо в мене йде перевірка спершу на те чи взагалі текст не дорівнює nil
if textView.text != "Нотатки" && (textView.text != nil) != textView.text.isEmpty{  - ц екастильне вирішення валідіції, проте як є

Reply:
перевірка на пустоту текста - це окреме обговорення. його взагалі краще замінити на guard)

я ж кажу за те, що ти в залежності від значення statusData.isOn або передаєш значення, або не передаєш його. тому я пропоную його так переписати:
```suggestion
let newDate: Date?
if statusData.isOn {
    newDate = dataValue
} else {
    newDate = nil
}
ObjectStore.shared.add(note: Note(id: count, name: textView.text, isDone: defaultValue, deadlineDate: newValue))
```

ми просто рахуємо це значення "за дужками", і потім його один раз вставляємо

Reply:
тепер поняв за шо ти